Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using Debug = UnityEngine.Debug;
+
+public class WebCamSource : MonoBehaviour {
+  public Dropdown CameraDropdown;
+  public Dropdown ResolutionDropdown;
+  public Dropdown FpsDropdown;
+  public Dropdown FlipDropdown;
+  public string DeviceName;
+  public int Fps = 30;
+  public bool Flip = false;
+  [SerializeField] private RenderTexture _buffer;
+  private WebCamTexture _webCam;
+  private int _width = 1280;
+  private int _height = 720;
+  private List<string> _devicesName = new List<string>();
+  private List<string> _resolutionOpt = new List<string>(){"1280X720","1600X900","1920X1080"};
+  private List<string> _fpsOpt = new List<string>(){"30Hz","60Hz"};
+  private List<string> _flipOpt = new List<string>(){"默认","水平翻转"};
+  //Start button event
+  public void Play() {
+    if(_webCam == null) {
+      _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+      _webCam.Play();
+    }
+  }
+  //Stop button event
+  public void Stop() {
+    if(_webCam != null) {
+      _webCam.Stop();
+      _webCam = null;
+    }
+  }
+  //DropdownChange event
+  public void CameraDropdownChange(int value) {
+    //Debug.Log(_devicesName.ToArray()[value]);
+    if(_devicesName.ToArray()[value] != null) {
+      DeviceName = _devicesName.ToArray()[value];
+    }
+  }
+  //DropdownChange event
+  public void ResolutionDropdownChange(int value) {
+    string _resolution = _resolutionOpt.ToArray()[value];
+    if(_resolution != null) {
+      switch(_resolution) {
+        case "1280X720":
```

Comment:
Do not base on string literal to determine the current choice of a dropdown list. Consider this case: if you want to change the choice from "1280X720" to a new text like "1280x720", you need to update every location of this string literal in the code.

Recommended solutions:

1. Use const variables to hold those string literals.
2. Define an Enum to hold those choices and uses a Dictionary to maintain the mapping from enums to string literals.
3. Use the int value got from the dropdown change event as the key, and maintain a Dictionary for the mapping from keys to string literals.